sdk/build.gradle.kts

```diff
@@ -72,6 +72,7 @@ dependencies {
   api(project(":plugin-annotation"))
   api(project(":extension-localization"))
   api(project(":plugin-lifecycle"))
+  compileOnly(Dependencies.asyncInflater)
```

Comment:
why compileOnly?

Reply:
thanks for the question!
we want to support in our API ability to inflate views async however at the same time we don't want to add even small binary size to our SDK module.
that's why we enforce users to provide async inflater instance as parameter in our API (which is possible only if they indeed have async layout inflater dependency) but our code recognizes it because of `compileOnly` (but we do not provide dependency impl as part of our SDK).

Reply:
do we have some numbers on how much binary size the async inflater takes?
Imho if our code relies on it to work(and it's not optional in the public interface) we should use implement dependency. It's a bit different from other cases like LocationEngine, where GoogleLocationService is optional and we have a safe fallback. wdyt? cc @mapbox/maps-android

Reply:
@pengdev I don't agree here.
Although we don't provide impl having API the way it is right now:
```
fun addViewAnnotation(
    @LayoutRes resId: Int,
    options: ViewAnnotationOptions,
    asyncInflater: AsyncLayoutInflater,
    asyncInflateCallback: (View) -> Unit
  )
```
actually _enforces_ user to add this dependency to call this API. There's no even a concept of safe fallback here because this API could not simply be called by user if he does not import dependency.

At the same time it's convenient possibility to move inflating off the main thread for advanced user cases with bunch of view annotations.

Reply:
Cool idea of saving binary size but I do want to make sure we provide a good developer experience. What happens if the developer calls into the codepath but didn't add the depdency on their end? ClassNotFoundException? 
If yes, I do think we can optimize the experience by catching this exception and throwing a new depedency missing exception that clearly indicates what a user can do to resolve this issue.

Reply:
IMHO well written javadoc here should prevent the possible confusion
Just need to check that it indeed won't compile without explicit dependency declaration on used end instead of runtime exception

Reply:
Discussed offline, will go with current solution.